Source/WebCore/xml/XSLTProcessorLibxslt.cpp

```diff
@@ -162,39 +162,12 @@ static inline void setXSLTLoadCallBack(xsltDocLoaderFunc func, XSLTProcessor* pr
 
 static int writeToStringBuilder(void* context, const char* buffer, int length)
 {
-    StringBuilder& resultOutput = *static_cast<StringBuilder*>(context);
-
-    // FIXME: Consider ways to make this more efficient by moving it into a
-    // StringBuilder::appendUTF8 function, and then optimizing to not need a
-    // Vector<UChar> and possibly optimize cases that can produce 8-bit Latin-1
-    // strings, but that would need to be sophisticated about not processing
-    // trailing incomplete sequences and communicating that to the caller.
-
-    Vector<UChar> outputBuffer(length);
-
-    UBool error = false;
-    int inputOffset = 0;
-    int outputOffset = 0;
-    while (inputOffset < length) {
-        UChar32 character;
-        int nextInputOffset = inputOffset;
-        U8_NEXT(reinterpret_cast<const uint8_t*>(buffer), nextInputOffset, length, character);
-        if (character < 0) {
-            if (nextInputOffset == length)
-                break;
-            ASSERT_NOT_REACHED();
-            return -1;
-        }
-        inputOffset = nextInputOffset;
-        U16_APPEND(outputBuffer.data(), outputOffset, length, character, error);
-        if (error) {
-            ASSERT_NOT_REACHED();
-            return -1;
-        }
-    }
-
-    resultOutput.appendCharacters(outputBuffer.data(), outputOffset);
-    return inputOffset;
+    auto& builder = *static_cast<StringBuilder*>(context);
+    FromUTF8 adapter(buffer, length);
+    if (adapter.conversionFailed)
+        return -1;
```

Comment:
Should bring back the `ASSERT_NOT_REACHED` here.

Reply:
This is also not in the latest PR.

Reply:
Was in the final one,